### DISCUSSION TITLE
How many genes necessary to be identfied as a pathway

### DISCUSSION THREAD

**Discussion by NGuslund** [reactome/ReactomeGSA#50] [General]
Hi, thank you for this package and the vignette online, very helpful.

I have two questions:
1) What is the minimum number of genes that need to be involved for the function to assign a pathway? And how can I see how many genes have been included in the 'calling' of a pathway. For example, when I am following this vignette (https://bioconductor.org/packages/release/bioc/vignettes/ReactomeGSA/inst/doc/analysing-scRNAseq.html) and plot_gsva_pathway(gsva_result, pathway_id = rownames(max_difference)[1]) gives my top pathway as "R-HSA-3296197". How confident can I be that this pathway is indeed involved in what my cells are doing? How many genes have correctly matched this pathway? I hope that makes sense! :)

2) I am doing an immune challenge over a timeline. So the 'clusters' that I am comparing against one another are the same cell types (i.e. B cells) sampled at different times. Is this still an appropriate analysis to be using to track pathway changes? If not do you have any suggestions?

Thank you for your time,
Naomi

**Comment by jgriss** [comment]
Hi @NGuslund ,

Thanks a lot for the very nice feedback!

1. That's very difficult to answer and essentially depends on your dataset. If you use `PADOG` genes found in many pathways are already down-weighted. Therefore, the top significant pathways are generally relevant to your dataset.

In `ssGSEA` as it's used in the scRNA-seq data this is not the case. You simply get the gene level information mapped on the pathway level. Therefore, you need some additional statistics to decide whether the pathway is relevant to your biological question.

Unfortunately, we currently do not report the number of genes per pathway for `ssGSEA` results but I've already added it to our ToDo list (#53).

2. Here, you can create and id for your cells consisting of the cell type + treatment. Then, you manually calculate the average gene expression per pathway using f.e. Seurat's inbuilt function. This pseudo-bulk data can then be analysed in `ReactomeGSA` where you can compare your timepoints or challenges.

**Reply by NGuslund** [reply]
Thank you for your quick and helpful response, I really appreciate it- analyzing data can be a lonely task sometimes!

I see in the 2020 paper 'ReactomeGSA - Efficient Multi-Omics Comparative Pathway Analysis', that in Figure 4 you show a heatmap, and you only show significant pathways (FDR < 0.1). Where can I find the significance of the pathways? In your tutorial https://bioconductor.org/packages/release/bioc/vignettes/ReactomeGSA/inst/doc/analysing-scRNAseq.html You do some commands to find the maximum differently expressed pathway. For example this is for my data:

![image](https://user-images.githubusercontent.com/85682207/139387419-2ff818e6-3ba1-4f4b-95cd-e99c9494380c.png)

Can I use the max difference to infer significance?

BW,
Naomi

**Reply by NGuslund** [reply]
How did you generate your supplementary data 1 in this paper?

**Reply by NGuslund** [reply]
Sorry for my multiple messages! I guess that you used PADOG? Would it be possible to share a script using PADOG? You write that PADOG is compatible with ReactomeGSA. (I am using seurat objects)

**Reply by jgriss** [reply]
Hi @NGuslund 

## Figure 4

You can find the complete code to recreate figure 4 from our manuscript in [this tutorial](https://github.com/reactome/ReactomeGSA-tutorials/blob/master/notebooks/TCGA_B_Cell_Analysis.ipynb)

## ssGSEA pathway significance

The ssGSEA based abundance values do not allow any conclusions about the *significance* of a pathway. You need to perform some additional statistical test based on your hypothesis to get there. These are simply abundance values.

## PADOG + Seurat

We are working on a new tutorial that covers this as well. In short, it very much depends on your biological question. The steps are as follows:

  1. Create pseudo-bulk RNA-seq datasets using Seurat's `AverageExpression` function
  2. Use these pseudo-bulk RNA-seq values as input to ReactomeGSA, treating them as individual samples

The way you create the average expression matrix is key and has to reflect the comparison you want to perform.

Hope this helps!
